src/components/OTable.vue

```diff
@@ -0,0 +1,245 @@
+            <slot name="controls-header__end-before" />
+
+
+            <template v-if="!hideSearch">
+              <slot name="controls-header__search">
+                <SearchInput v-model="search" :disabled="disableSearch" :placeholder="searchPlaceholder" :label="searchLabel" />
+              </slot>
+            </template>
+
+            <slot name="controls-header__end-after" />
+          </slot>
+        </div>
+      </slot>
+    </div>
+
+    <p-table v-bind="attrs" :data="_data" :columns="_columns">
+      <template #select-heading>
+        <p-checkbox v-if="!noData" v-model="select" :disabled="disableSelect" />
+        <span v-else />
+      </template>
+
+      <template #actions-heading>
+        <span />
+      </template>
+
+      <template #select="{ row }">
+        <p-checkbox v-model="_selectedRows" :value="row" :disabled="disableSelect" />
+      </template>
+
+      <template v-for="(_, name) in $slots" #[name]="slotData">
+        <slot :name="name" v-bind="slotData" />
+      </template>
+
+      <template #actions="{ row }">
+        <div class="o-table__row__actions">
+          <slot name="actions" :row="row" />
+        </div>
+      </template>
+
+
+      <template #empty-state>
+        <p-empty-results>
+          <template #message>
+            No {{ emptyResultsLabel }}
+          </template>
+          <template #actions>
+            <p-button v-if="!noData" size="sm" secondary @click="clear">
+              Clear Filters
+            </p-button>
+          </template>
+        </p-empty-results>
+      </template>
+    </p-table>
+
+    <div class="o-table__controls-footer">
+      <slot name="controls__footer">
+        <div class="o-table__controls__section o-table__controls__section-start">
+          <slot name="controls__footer__start" />
+        </div>
+
+        <div class="o-table__controls__section o-table__controls__section-end">
+          <slot name="controls__footer__end" />
+        </div>
+      </slot>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+  export default {
+    name: 'OTable',
+    expose: [],
+    inheritAttrs: false,
+  }
+</script>
+
+<script lang="ts" setup>
+  import { TableColumn, TableData } from '@prefecthq/prefect-design'
+  import { ref, useAttrs, computed } from 'vue'
+  import { SearchInput, ResultsCount, SelectedCount } from '@/components'
+
+  export type TableDataFilter = (term: string, row: TableData, index: number, arr: TableData[]) => boolean
+
+  const attrs = useAttrs()
+
+  const props = withDefaults(defineProps<{
+    selectedRows?: TableData[],
+    columns: TableColumn[],
+    data: TableData[],
+    emptyResultsLabel?: string,
+    searchPlaceholder?: string,
+    searchLabel?: string,
+    rowsCountLabel?: string,
+    disableSelect?: boolean,
+    disableSearch?: boolean,
+    hideSearch?: boolean,
+    hideSelectColumn?: boolean,
+    hideActionsColumn?: boolean,
+    search?: TableDataFilter,
```

Comment:
pleek and I just spent some time talking over something very similar in p-select. We came to the conclusion that passing in `data`, and a method to filter that `data` was not ideal, and that the parent component should do the filtering of `data`. I'm not 100% sure the implications of that change here, maybe this component should v-model the search value or emit "search"?

here's the related PR that removes filter prop from p-select
https://github.com/PrefectHQ/prefect-design/pull/561

Reply:
+1 here. I'd rather not "configure" a component but rather express things as code. 

I think search needs to be a v-model anyway to support api search.

Reply:
Agreed with this - updated this component to use a model value for search and data to allow API searching/filtering and to emit a `clear` event when the `Clear filters` button is clicked; that will allow consumers to add as many filters as needed and to handle those external to the component.